src/Core/Grid/Data/Factory/ManufacturerAddressGridDataFactory.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Data\Factory;
+
+use PrestaShop\PrestaShop\Core\Grid\Data\GridData;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+use PrestaShop\PrestaShop\Core\Grid\Record\RecordCollection;
+
+/**
+ * Gets data for manufacturer addresses grid
+ */
+class ManufacturerAddressGridDataFactory implements GridDataFactoryInterface
+{
+    /**
+     * @var GridDataFactoryInterface
+     */
+    private $manufacturerAddressDataFactory;
+
+    /**
+     * @param GridDataFactoryInterface $manufacturerAddressDataFactory
+     */
+    public function __construct(GridDataFactoryInterface $manufacturerAddressDataFactory)
+    {
+        $this->manufacturerAddressDataFactory = $manufacturerAddressDataFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getData(SearchCriteriaInterface $searchCriteria)
+    {
+        $addresses = $this->manufacturerAddressDataFactory->getData($searchCriteria);
+
+        $modifiedRecords = $this->applyModification(
+            $addresses->getRecords()->all()
+        );
+
+        return new GridData(
+            new RecordCollection($modifiedRecords),
+            $addresses->getRecordsTotal(),
+            $addresses->getQuery()
+        );
+    }
+
+    /**
+     * @param array $addresses
+     *
+     * @return array
+     */
+    private function applyModification(array $addresses)
```

Comment:
wtf? why are we doing private changes here so the PrestaShop developers won't be able to change or alter them?

Reply:
needs to be configurable in the next minors versions (like the applicators introduced by @eternoendless :+1: )

Reply:
I think developers can already do it, but we can't, that's why be need a decorator.

Reply:
@mickaelandrieu I dont understand the issue o_O. If a developer wants to alter this behavior, he can:
- override the Symfony service with another service
- use another decorator to add his behavior

Did you want the function to be `protected` ? `public` ? Then ` getData(SearchCriteriaInterface $searchCriteria)` is already public, why do we need `applyModification()` to be opened as well ?

Reply:
> override the Symfony service with another service

it should be done via a hook, remember that Symfony provides an extra experience, not a replacement (in 1.7).

> use another decorator to add his behavior

what do you mean? to be able to decorate the modifications applied to the data you should be able to have the hand on the " data modifiers" using a hook: this is what I'm suggesting here. 

> I think developers can already do it (via hooks)

I may be wrong but the current hook is called after this function call, so the developers won't have access to the unaltered data, isn't it?

Reply:
> I may be wrong but the current hook is called after this function call, so the developers won't have access to the unaltered data, isn't it?

Yep, I think you're right.

Some time ago I wanted to introduce `Modifiers`, that could be hooked in (by registering a service & tagging it) for any Grid by either module developers or core developer.

Reply:
> I may be wrong but the current hook is called after this function call, so the developers won't have access to the unaltered data, isn't it?

Weird.. We must be able to alter data when it hasn't been altered :thinking: 
Should be important to add a hook here.